compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -70,20 +70,24 @@ pub(crate) enum ForLifetimeSpanType {
     BoundTail,
     TypeEmpty,
     TypeTail,
+    ClosureEmpty,
+    ClosureTail,
 }
 
 impl ForLifetimeSpanType {
     pub(crate) fn descr(&self) -> &'static str {
         match self {
             Self::BoundEmpty | Self::BoundTail => "bound",
             Self::TypeEmpty | Self::TypeTail => "type",
+            Self::ClosureEmpty | Self::ClosureTail => "closure",
         }
     }
 
     pub(crate) fn suggestion(&self, sugg: &str) -> String {
         match self {
             Self::BoundEmpty | Self::TypeEmpty => format!("for<{}> ", sugg),
-            Self::BoundTail | Self::TypeTail => format!(", {}", sugg),
+            Self::ClosureEmpty => format!("for<{}>", sugg),
```

Comment:
Could you merge this arm with the previous one?

Reply:
No, it doesn't have a trailing space.

Reply:
Why do you want to skip the trailing whitespace?  I couldn't find a ui test suggesting to introduce a closure for bound.

Reply:
Because we only suggest `for<'lifetime>` when there is already `for<>`. So we suggest replacing `for<>` with `for<'lifetime>`. In cases of `BoundEmpty` and `TypeEmpty` this is used when there is no `for<>` at all, so in these cases it makes sense to add a trailing space.

However at closer inspection... I don't think this is ever used. This variant is added to `missing_named_lifetime_spots` which is used only in [`add_missing_lifetime_specifiers_label`](https://github.com/rust-lang/rust/blob/27905df77fd6dd9ad6d73b7820ff3d020ec2e7b3/compiler/rustc_resolve/src/late/diagnostics.rs#L2265) that is only used in [`resolve_elided_lifetimes`](https://github.com/rust-lang/rust/blob/27905df77fd6dd9ad6d73b7820ff3d020ec2e7b3/compiler/rustc_resolve/src/late/lifetimes.rs#L2419) which, IIRC, is not called for closures.